src/net/socket.rs

```diff
@@ -61,12 +62,12 @@ impl Socket {
                         match self.event_sender.send(SocketEvent::Packet(packet)) {
                             Ok(_) => {}
                             Err(e) => {
-                                error!("Error sending SocketEvent: {:?}", e);
+                                error!("Error sending SocketEvent: {:?}", e.description());
                             }
                         };
                     }
                     None => {
-                        error!("Empty packet received");
+                        debug!("Empty packet received");
```

Comment:
Do we expect to ever receive an empty packet in this test? If not, it should probably be an error.

Reply:
We do and this should still be fixed, in case of fragmentation we could have None, however my other PR is removing a lot of this code and doesn't have to deal with this anymore.